packages/Filter/src/Filter.js

```diff
@@ -0,0 +1,143 @@
+import React from "react";
+import PropTypes from "prop-types";
+import AddIcon from "@paprika/icon/lib/Add";
+import Button from "@paprika/button";
+import FilterIcon from "@paprika/icon/lib/Filter";
+import Panel from "@paprika/panel";
+import useI18n from "@paprika/l10n/lib/useI18n";
+import Item from "./components/Item";
+import FilterContext from "./context";
+import columnShape from "./columnShape";
+import * as types from "./types";
+import rules, { defaultRulesByType, logicalFilterOperators } from "./rules";
+
+import * as sc from "./Filter.styles";
+
+function getLabelText(numberOfFilters, I18n) {
+  return numberOfFilters ? I18n.t("filter.lowercase_label", { count: numberOfFilters }) : I18n.t("filter.label");
+}
+
+export default function Filter(props) {
+  const {
+    numberApplied,
+    children,
+    columns,
+    data,
+    onAddFilter,
+    onApply,
+    onCancel,
+    onChangeOperator,
+    onClear,
+    operator,
+    rulesByType,
+  } = props;
+  const I18n = useI18n();
+  const filterRef = React.useRef(null);
+  const [isOpen, setIsOpen] = React.useState(false);
+
+  function handleClickTrigger() {
+    setIsOpen(prevIsOpen => !prevIsOpen);
+  }
+
+  function handleApply() {
+    onApply();
+    setIsOpen(false);
+  }
+
+  function handleCancel() {
+    setIsOpen(false);
+    onCancel();
+  }
+
+  function handleClear() {
+    onClear();
+  }
+
+  return (
+    <FilterContext.Provider value={{ filterRef, columns, data, operator, onChangeOperator, rulesByType }}>
+      <Panel
```

Comment:
🤔 debating with myself about if including the panel inside of the filter component or not, did you considered an option where you split this component in two, so the user assemble those by themselves but you provide them ready for use?

Something like

```js
import Filter, { Panel } from "@paprika/filter"

<Panel>
  <Filter />
</Panel>
```

An approach like this would decouple the panel from the filter, so the filter could actually render wherever it needs to render. 

Just a thought.

Reply:
I'm checking with Sarah if we have a case of using filters but not the panel. If not, I think the only benefit would be bringing the flexibility to the consumer app to control the props on `Panel` `Panel.Trigger`, that's a good point, because I forgot support `zIndex` in this PR. 

Other than that I don't think there're many values to ask them to write `<Filter.Item>` inside `<Filter>` and `<Filter>` inside `<Panel>`. It's only for the decoupling purpose like you said.

Reply:
Oh just got Sarah's comment, she said we don't have any case about using advanced filters outside the panel